assert/http_assertions.go

```diff
@@ -90,6 +90,29 @@ func HTTPError(t TestingT, handler http.HandlerFunc, method, url string, values
 	return isErrorCode
 }
 
+// HTTPStatusCode asserts that a specified handler returns a specified status code.
+//
+//  assert.HTTPStatusCode(t, myHandler, "GET", "/notImplemented", nil, 501)
+//
+// Returns whether the assertion was successful (true) or not (false).
+func HTTPStatusCode(t TestingT, handler http.HandlerFunc, method, url string, values url.Values, statuscode int, msgAndArgs ...interface{}) bool {
+	if h, ok := t.(tHelper); ok {
+		h.Helper()
+	}
+	code, err := httpCode(handler, method, url, values)
+	if err != nil {
+		Fail(t, fmt.Sprintf("Failed to build test request, got error: %s", err))
+		return false
+	}
```

Comment:
This logic is common to this function and at least the `HTTPRedirect`, `HTTPSuccess` and the `HTTPError` functions. Would it not make sense to extract them into one function and the reference them in there?

Reply:
Will take a look.

Reply:
@boyan-soubachov I looked at this and I agree this doesn't feel very DRY. However, wrapping the call to httpCode and the error handling into a separate function that has TestingT as an argument feel kinda silly to me... maybe that's because I'm new to Go, dunno.

I could handle the error checking in httpCode itself, but that would also require adding TestingT as an argument since we'd need it to Fail().

As for the h.Helper() code... I'm not entirely clear on what it does but that code's all over the place in Testify, so that makes no sense for me to extract from these particular functions.

I'd say leave it as it is now or move the error checking into httpCode()... please let me know what you want to do.

Reply:
Fair enough, having had a second look, I think you're correct. I think it's about as simple as it should be at the moment :)